samplejava/src/main/java/com/example/chattutorial/ChannelActivity4.java

```diff
@@ -65,12 +68,18 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
         MessageListViewModel messageListViewModel = provider.get(MessageListViewModel.class);
         MessageInputViewModel messageInputViewModel = provider.get(MessageInputViewModel.class);
 
-        // Set view factory for Imgur attachments
-        binding.messageListView.setAttachmentViewFactory(new ImgurAttachmentViewFactory());
+        // Set view factory manager for Imgur attachments
+        ImgurAttachmentFactory imgurAttachmentFactory = new ImgurAttachmentFactory();
+
+        List<ImgurAttachmentFactory> imgurAttachmentViewFactories = new ArrayList<ImgurAttachmentFactory>();
```

Comment:
Again, `List.of()` not supported by API 21 without desugaring, making this feel clunky.